sdk/src/types/block/address/bech.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::hash::Hash;
+
+use serde::{Deserialize, Serialize};
+
+use crate::types::block::address::Address;
+
+/// An address and its network type.
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub struct Bech32 {
+    pub(crate) inner: Address,
+    #[serde(rename = "bech32Hrp")]
+    pub(crate) bech32_hrp: String,
```

Comment:
I'd call it just hrp since it's already within something called bech32

Reply:
shall i rename inner to address and bech32_ to just hrp?

Reply:
> inner to address

As you want, renaming to address may end up in situations like `address.address`

> bech32_ to just hrp

yes pls